### PR TITLE
PT-2241 - fix(pt-online-schema-change): using --skip-check-slave-lag multiple times does not work as expected

### PR DESCRIPTION
**Pull request by sgjr**
using `--skip-check-slave-lag` multiple times does not work as expected.

- [x] The contributed code is licensed under GPL v2.0
- [x] Contributor Licence Agreement (CLA) is signed
- [x] util/update-modules has been ran
- [x] Documention updated
- [x] Test suite update

### How to reproduce?

> I have three slaves, slave01-03, and slave03 has replication delay. When using the --skip-check-slave-lag option of pt-online-schema-change to skip the replication delay check for slave02 and slave03, it still waits for slave03 to catch up.

```
pt-online-schema-change --user=ddl_user --password=xxxxx --host=$host   P=3306,D=test,t=t6 --max-lag=1  --max-load="Threads_running=500" --critical-load="Threads_running=2048" --alter-foreign-keys-method=rebuild_constraints --no-check-unique-key-change --chunk-size=1000 --alter 'add column col052305 varchar(20) not null default "abc"'     --skip-check-slave-lag h=$slave02_host,P=3306  --skip-check-slave-lag h=$slave03_host,P=3306  --execute
```

image

### CODE REVIEW

**Review by svetasmirnova — CHANGES_REQUESTED** [review]
Please add test cases for this situation. You will find a how-to in the Contribution guide and examples in the t/pt-online-schema-change directory.

**Comment by svetasmirnova** [comment]
Updated PR header, so all updates are linked to the bug report.

**Comment by sgjr** [comment]
> Please add test cases for this situation. You will find a how-to in the Contribution guide and examples in the t/pt-online-schema-change directory.

The test cases has been added. But I can't modify my branch name. Do I need to create a new standard branch and resubmit the merge request?

**Comment by svetasmirnova** [comment]
Thank you, I will review your changes.

No need to rename the branch: automation work with commit messages and PR subject.

File: t/pt-online-schema-change/pt-2241.t [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+unlike(
+      $output,
+      qr/Replica lag is \d+ seconds on .*  Waiting/s,
+      "--skip-check-slave-lag is really skipping the slave",
+);
+
+# #############################################################################
+# Done.
+# #############################################################################
```
Comment (svetasmirnova) [inline t/pt-online-schema-change/pt-2241.t:97]:
```suggestion
# #############################################################################
diag("Setting slave delay to 0 seconds");
$slave1_dbh->do('STOP SLAVE');
$slave2_dbh->do('STOP SLAVE');
$master_dbh->do("RESET MASTER");
$slave1_dbh->do('RESET SLAVE');
$slave1_dbh->do('START SLAVE');
$slave2_dbh->do('RESET SLAVE');
$slave2_dbh->do('START SLAVE');

```

Comment (svetasmirnova) [reply t/pt-online-schema-change/pt-2241.t:97]: We need to reset the delay on replicas, otherwise, all subsequent tests will be slow.

**Review by svetasmirnova — CHANGES_REQUESTED** [review]
Please apply the requested changes.

**Review by svetasmirnova — APPROVED** [review]

**Review by EvgeniyPatlan — APPROVED** [review]
LGTM :+1:

**Comment by svetasmirnova** [comment]
@sgjr your PR is technically approved but we cannot accept it yet due to CLA issue:

> Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.percona.com/percona/percona-toolkit?pullRequest=622) before we can accept your contribution.
> 1 out of 2 committers have signed the CLA.
> 
> ✅ sgjr
> x caopeng
> caopeng seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).
> You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.percona.com/check/percona/percona-toolkit?pullRequest=622) it.

Could you please ask caopeng to sign up for a GitHub account with the address, used for the commit, and sign the license?

Thank you!

**Comment by Aleksa-del** [comment]
Hello, @sgjr ! Thank you for your contributions! We would like to send you a gift from Percona. Please, contact us at community-team@percona.com.

**Pull request merged**